Neos.Flow/Classes/ObjectManagement/CompileTimeObjectManager.php

```diff
@@ -228,7 +228,10 @@ protected function registerClassFiles(array $packages)
                 }
                 if ($package instanceof FlowPackageInterface && $shouldRegisterFunctionalTestClasses) {
                     foreach ($package->getFunctionalTestsClassFiles() as $fullClassName => $path) {
-                        if (PHP_MAJOR_VERSION < 8 && strpos($fullClassName, '\\PHP8\\') !== false) {
+                        if (version_compare(PHP_VERSION, '8.0', '<=') && strpos($fullClassName, '\\PHP8\\') !== false) {
```

Comment:
Not sure if it's much better, but my PHPStorm suggests to change this line to
```suggestion
                        if (PHP_VERSION_ID <= 80000 && strpos($fullClassName, '\\PHP8\\') !== false) {
```

But apart from that: I'm wondering what these checks actually do?
Could we add some inline comment to explain that?